essentials-framework/Essentials Core/PepperDashEssentialsBase/Queues/GenericQueue.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using Crestron.SimplSharp;
+using Crestron.SimplSharpPro.CrestronThread;
+using PepperDash.Core;
+
+namespace PepperDash_Essentials_Core.Queues
+{
+    /// <summary>
+    /// Threadsafe processing of queued items with pacing if required
+    /// </summary>
+    public class GenericQueue : IQueue<IQueueMessage>
+    {
+        private readonly string _key;
+        protected readonly CrestronQueue<IQueueMessage> _queue;
+        protected readonly Thread _worker;
+        protected readonly CEvent _waitHandle = new CEvent();
+        
+        private readonly bool _delayEnabled;
+        private readonly int _delayTime;
+
+        /// <summary>
+        /// If the instance has been disposed.
+        /// </summary>
+        public bool Disposed { get; private set; }
+
+        /// <summary>
+        /// Constructor for generic queue with no pacing
+        /// </summary>
+        /// <param name="key">Key</param>
+        public GenericQueue(string key)
```

Comment:
We should add another constructor with an option for queue size. Crestron Queues default to 5, and don't automatically increase their size